server/api_admin.go

```diff
@@ -7,11 +7,138 @@ import (
 	"net/http"
 
 	"errors"
+	"fmt"
 
 	"github.com/gin-gonic/gin"
+	"github.com/mattermost/mattermost-plugin-ai/server/embeddings"
 	"github.com/mattermost/mattermost/server/public/model"
 )
 
+const defaultBatchSize = 100
+
+func (p *Plugin) handleReindexPosts(c *gin.Context) {
+	type PostRecord struct {
+		ID       string `db:"id"`
+		Message  string `db:"message"`
+		UserID   string `db:"userid"`
+		CreateAt int64  `db:"createat"`
+		TeamID   string `db:"teamid"`
+
+		ChannelID   string `db:"channelid"`
+		ChannelName string `db:"channelname"`
+		ChannelType string `db:"channeltype"`
+	}
+
+	// Check if search is initialized
+	if p.search == nil {
+		c.AbortWithError(http.StatusBadRequest, fmt.Errorf("search functionality is not configured"))
+		return
+	}
+
+	var posts []PostRecord
+	lastCreateAt := int64(0)
+	lastID := ""
+
+	if err := p.search.Clear(c.Request.Context()); err != nil {
+		c.AbortWithError(http.StatusInternalServerError, fmt.Errorf("failed to clear search index: %w", err))
+		return
+	}
+	newSearch, err := p.initSearch()
+	if err != nil {
+		c.AbortWithError(http.StatusInternalServerError, fmt.Errorf("failed to re-initialize search: %w", err))
+		return
+	}
+	p.search = newSearch
+
+	for {
+		query := `SELECT
```

Comment:
nit: should this query be moved to the search store to keep the same level of abstraction for the whole code block?

Reply:
Actually no, because it doesn't actually relate to the embedding search. It's getting the posts from the MM server in order to index them and is generic to any `EmbeddingSearch` service.

I could put it int the server/store.go file, but actually I am kinda fond of having everything in one place. Not sure on that one though.